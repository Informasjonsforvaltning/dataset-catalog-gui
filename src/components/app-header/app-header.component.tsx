import React from 'react';
import { compose, withProps } from 'recompose';

import localization from '../../services/localization';
import './app-header.scss';
import { authService } from '../../services/auth/auth-service';
import { getConfig } from '../../config';
import { User } from '../../lib/auth/auth';

interface Props {
  user: User;
}

export const AppHeaderPure = ({ user }: Props) => {
  const logoImagePath = getConfig().useDemoLogo
    ? '/img/logo-registrering-demo.svg'
    : '/img/logo-registrering.svg';

  return (
    <header>
      <div>
        <a
          id="focus-element"
          className="uu-invisible"
          href={`${location.pathname}#content`}
          aria-hidden="true"
        >
          {localization.app.skipLink}
        </a>
      </div>
      <div id="skip-link-wrap">
        <a id="skip-link" href={`${location.pathname}#content`}>
          {localization.app.skipLink}
        </a>
      </div>
      <div className="fdk-header">
        <div className="container">
          <div className="row">
            <div className="col-6 col-md-4">
              <a title="Link til Felles datakatalog" href="/">
                <span className="uu-invisible" aria-hidden="false">
                  {localization.app.navigateFrontpage}
                </span>
                <img
                  className="fdk-logo"
                  src={logoImagePath}
                  alt="Logo for Felles datakatalog"
                />
              </a>
            </div>
            <div className="col-6 col-md-8 d-flex align-items-center fdk-header-text_items justify-content-end">
              {user && user.name && (
                <div className="mr-4">
                  <i className="fa fa-user fdk-fa-left fdk-color-primary" />
                  {user.name}
                </div>
              )}
              {user && (
                <div className="mr-4 fdk-auth-link">
                  <button
                    onClick={() => authService.logout()}
                    type="button"
                    className="fdk-btn-no-border"
                  >
                    {localization.app.logOut}
                  </button>
                </div>
              )}
            </div>
          </div>
        </div>
      </div>
    </header>
  );
};

const enhance = compose(withProps(() => ({ user: authService.getUser() })));

export const AppHeader = enhance(AppHeaderPure);
