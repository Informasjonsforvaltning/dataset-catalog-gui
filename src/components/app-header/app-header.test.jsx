import React from 'react';
import { shallow } from 'enzyme';
import { AppHeaderPure } from './app-header.component';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {
    user: { name: 'john doe' }
  };
  wrapper = shallow(<AppHeaderPure {...defaultProps} />);
});

test('should render AppHeader correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
