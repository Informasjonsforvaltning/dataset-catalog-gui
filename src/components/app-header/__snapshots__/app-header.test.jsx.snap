// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AppHeader correctly 1`] = `
<header>
  <div>
    <a
      aria-hidden="true"
      className="uu-invisible"
      href="/#content"
      id="focus-element"
    >
      Hopp til hovedinnhold
    </a>
  </div>
  <div
    id="skip-link-wrap"
  >
    <a
      href="/#content"
      id="skip-link"
    >
      Hopp til hovedinnhold
    </a>
  </div>
  <div
    className="fdk-header"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-6 col-md-4"
        >
          <a
            href="/"
            title="Link til Felles datakatalog"
          >
            <span
              aria-hidden="false"
              className="uu-invisible"
            >
              GÃ¥ til forside
            </span>
            <img
              alt="Logo for Felles datakatalog"
              className="fdk-logo"
              src="/img/logo-registrering.svg"
            />
          </a>
        </div>
        <div
          className="col-6 col-md-8 d-flex align-items-center fdk-header-text_items justify-content-end"
        >
          <div
            className="mr-4"
          >
            <i
              className="fa fa-user fdk-fa-left fdk-color-primary"
            />
            john doe
          </div>
          <div
            className="mr-4 fdk-auth-link"
          >
            <button
              className="fdk-btn-no-border"
              onClick={[Function]}
              type="button"
            >
              Logg ut
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
